extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/sql/HllSketchBaseSqlAggregator.java

```diff
@@ -193,7 +193,8 @@ public Aggregation toDruidAggregation(
             tgtHllType,
             stringEncoding,
             finalizeSketch || SketchQueryContext.isFinalizeOuterSketches(plannerContext),
-            ROUND
+            ROUND,
+            false
```

Comment:
oops, i meant to wire these up to the type of the input, if it is an array type it should set the flag to true, if not it should set to false

Reply:
i've correctly wired this up to the sql agg functions so it will be set if the input type is ARRAY typed, as well as added tests that are equivalent to the native ingestion and query tests

Reply:
Do we have clear documentation on what type a column is expected to be when, e.g., it was ingested as a long, then ingested as a String and then an ARRAY and then back to a String?  I'm a bit concerned with relying on the expected type from the SQL schema as it's not guaranteed to accurately reflect what is actually in the data.  I think it's better to have the query itself define what it wants to do and then use the information from the Sql schema as hints to maybe specialize that (but the behavior defined by the actual written query always wins).

Reply:
Not sure how clearly it is documented, but the current default behavior is that the schema is the 'least restrictive' type https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/column/ColumnType.java#L154, which means if any segment has ARRAY the column will be ARRAY typed.